src/main/java/edu/greenblitz/robotName/subsystems/gyros/Pigeon2Gyro.java

```diff
@@ -26,24 +27,24 @@ public Pigeon2Gyro(int deviceID) {
 	
 	@Override
 	public void updateYaw(Rotation2d yaw) {
-		yawOffset.minus(yaw.plus(Rotation2d.fromRadians(lastInputs.yaw)));
+		pigeon2.setYaw(yaw.getDegrees());
 	}
 	
 	@Override
 	public void updatePitch(Rotation2d pitch) {
-		pitchOffset.plus(pitch.plus(Rotation2d.fromRadians(lastInputs.pitch)));
+		pitchOffset = Rotation2d.fromDegrees(pigeon2.getPitch().getValue() - pitch.getDegrees());
 	}
 	
 	@Override
 	public void updateRoll(Rotation2d roll) {
-		rollOffset.plus(roll.plus(Rotation2d.fromRadians(lastInputs.roll)));
+		rollOffset = Rotation2d.fromDegrees(pigeon2.getRoll().getValue() - roll.getDegrees());
 	}
 	
 	@Override
 	public void updateInputs(GyroInputsAutoLogged inputs) {
-		inputs.yaw = (Math.toRadians(pigeon2.getYaw().getValue()) - yawOffset.getRadians()) % (2 * Math.PI);
-		inputs.pitch = (Math.toRadians(pigeon2.getPitch().getValue()) - pitchOffset.getRadians()) % (2 * Math.PI);
-		inputs.roll = (Math.toRadians(pigeon2.getRoll().getValue()) - rollOffset.getRadians()) % (2 * Math.PI);
+		inputs.yaw = (Units.degreesToRadians(pigeon2.getYaw().getValue()) - yawOffset.getRadians()) % GyroConstants.CIRCLE_CIRCUMFRENCE.getRadians();
```

Comment:
why..? this is unneeded, the previous version was fine just keep the Units.degToRad like you did here because its more informative